### DISCUSSION TITLE
Welcome to TfsCmdlets Discussions!

### DISCUSSION THREAD

**Discussion by igoravl** [igoravl/TfsCmdlets#183] [Announcements]
## 👋 Welcome!

  We’re using Discussions as a place to connect with other members of our community. We hope that you:

  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by sbrassard** [comment]
I'm curious about Get-TfsWorkItem and the -Fields property. The documentation says to specify a '*' to return all fields. Any ideas?

Get-TfsWorkItem -AreaPath $areaPath -WorkItemType "User Story" -Fields "*"

Get-TfsWorkItem : Value cannot be null.
Parameter name: values
At line:1 char:1
+ Get-TfsWorkItem -AreaPath $areaPath -WorkItemType "User Story" -F ...
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (:) [Get-TfsWorkItem], ArgumentNullException
    + FullyQualifiedErrorId : ArgumentNullException,TfsCmdlets.Cmdlets.WorkItem.GetWorkItem

**Reply by igoravl** [reply]
It looks like a bug. Thanks for reporting.

**Reply by sbrassard** [reply]
Igor - There may be another bug in the Get-TfsWorkItem method. When the number of items return more than 200, it throws an error. 

Get-TfsWorkItem -AreaPath $areaPath -WorkItemType "User Story" 
Get-TfsWorkItem -AreaPath $areaPath -Type "User Story" 

Both result in the following error:

Get-TfsWorkItem : Error getting work items (Not Found. )
At line:1 char:1
+ Get-TfsWorkItem -AreaPath $TeamDetails.AreaPath -Type "User Story"
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (:) [Get-TfsWorkItem], Exception
    + FullyQualifiedErrorId : VssServiceResponseException,TfsCmdlets.Cmdlets.WorkItem.GetWorkItem


And

Get-TfsWorkItem : Error getting work items (Not Found. )
At line:1 char:1
+ Get-TfsWorkItem -AreaPath $TeamDetails.AreaPath -WorkItemType "User S ...
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (:) [Get-TfsWorkItem], Exception
    + FullyQualifiedErrorId : VssServiceResponseException,TfsCmdlets.Cmdlets.WorkItem.GetWorkItem


However, Removing the "Type" or "WorkItemType" property, will result in returning a full list:

PS C:\Users\myuser> Get-TfsWorkItem -AreaPath $TeamDetails.AreaPath
WARNING: Your query resulted in 243 work items, therefore items must be fetched one at a time. This may take a
while. For best performance, write queries that return less than 200 items.

Id       Rev  Type                 State           Changed              Assigned To               Title
--       ---  ----                 -----           -------              -----------               -----
370674   22   User Story           Closed          2/8/2023 1:56:45 PM                            AIX Upgrade f...
370675   8    Initiative           In Progress     1/24/2023 5:04:59 PM                           2023 Datacent...
370676   11   User Story           New             1/19/2023 1:52:09 PM                           VMware vCente...
370677   12   User Story           Active          1/27/2023 1:51:38 PM                           Pure Storage ...
...
...

The following is running the same command with both -Debug and -Verbose properties:

PS C:\Users\myuser> Get-TfsWorkItem -AreaPath $TeamDetails.AreaPath -WorkItemType "User Story" -Debug -Verbose
VERBOSE: [09:02:40.1286] [Get-TfsWorkItem] BeginProcessing
VERBOSE: [09:02:40.1286] [Get-TfsWorkItem] ProcessRecord
VERBOSE: [09:02:40.1569] [Get-TfsWorkItem] Running [Get-TfsWorkItem] with parameter set 'Simple query' and the
following implicit and explicit arguments:
VERBOSE: [09:02:40.1579] [Get-TfsWorkItem] {"AreaPath" = "IT Main\\Cloud Infrastructure"; "WorkItemType" = ["User
Story"]; "BoardColumnDone" = false; "Ever" = false; "Revision" = 0; "AsOf" = "0001-01-01T00:00:00"; "Fields" =
["System.AreaPath"; "System.TeamProject"; "System.IterationPath"; "System.WorkItemType"; "System.State";
"System.Reason"; "System.CreatedDate"; "System.CreatedBy"; "System.ChangedDate"; "System.ChangedBy";
"System.CommentCount"; "System.Title"; "System.BoardColumn"; "System.BoardColumnDone";
"Microsoft.VSTS.Common.StateChangeDate"; "Microsoft.VSTS.Common.Priority"; "Microsoft.VSTS.Common.ValueArea";
"System.Description"; "System.Tags"]; "TimePrecision" = false; "ShowWindow" = false; "Deleted" = false;
"IncludeLinks" = false}
VERBOSE: [09:02:40.1619] [Get-TfsWorkItem] Running [Get-TfsWorkItem] with parameter set 'Simple query' and the
following implicit and explicit arguments:
VERBOSE: [09:02:40.1634] [Get-TfsWorkItem] {"AreaPath" = "IT Main\\Cloud Infrastructure"; "WorkItemType" = ["User
Story"]; "BoardColumnDone" = false; "Ever" = false; "Revision" = 0; "AsOf" = "0001-01-01T00:00:00"; "Fields" =
["System.AreaPath"; "System.TeamProject"; "System.IterationPath"; "System.WorkItemType"; "System.State";
"System.Reason"; "System.CreatedDate"; "System.CreatedBy"; "System.ChangedDate"; "System.ChangedBy";
"System.CommentCount"; "System.Title"; "System.BoardColumn"; "System.BoardColumnDone";
"Microsoft.VSTS.Common.StateChangeDate"; "Microsoft.VSTS.Common.Priority"; "Microsoft.VSTS.Common.ValueArea";
"System.Description"; "System.Tags"]; "TimePrecision" = false; "ShowWindow" = false; "Deleted" = false;
"IncludeLinks" = false}
VERBOSE: [09:02:40.1674] [Get-TfsWorkItem] Get currently connected Collection
VERBOSE: [09:02:40.1684] [Get-TfsWorkItem] GetClient: Getting an instance of
[Microsoft.TeamFoundation.WorkItemTracking.WebApi.WorkItemTrackingHttpClient]
VERBOSE: [09:02:40.1694] [Get-TfsWorkItem] Running [Get-TfsTeamProject] with parameter set 'Simple query' and the
following implicit and explicit arguments:
VERBOSE: [09:02:40.1704] [Get-TfsWorkItem] {"AreaPath" = "IT Main\\Cloud Infrastructure"; "WorkItemType" = ["User
Story"]; "BoardColumnDone" = false; "Ever" = false; "Revision" = 0; "AsOf" = "0001-01-01T00:00:00"; "Fields" =
["System.AreaPath"; "System.TeamProject"; "System.IterationPath"; "System.WorkItemType"; "System.State";
"System.Reason"; "System.CreatedDate"; "System.CreatedBy"; "System.ChangedDate"; "System.ChangedBy";
"System.CommentCount"; "System.Title"; "System.BoardColumn"; "System.BoardColumnDone";
"Microsoft.VSTS.Common.StateChangeDate"; "Microsoft.VSTS.Common.Priority"; "Microsoft.VSTS.Common.ValueArea";
"System.Description"; "System.Tags"]; "TimePrecision" = false; "ShowWindow" = false; "Deleted" = false;
"IncludeLinks" = false}
VERBOSE: [09:02:40.1744] [Get-TfsWorkItem] Get currently connected Collection
VERBOSE: [09:02:40.1744] [Get-TfsWorkItem] GetClient: Getting an instance of
[Microsoft.TeamFoundation.Core.WebApi.ProjectHttpClient]
VERBOSE: [09:02:40.1774] [Get-TfsWorkItem] Get currently connected team project
VERBOSE: [09:02:40.1774] [Get-TfsWorkItem] Simple query generated: "SELECT System.AreaPath, System.TeamProject,
System.IterationPath, System.WorkItemType, System.State, System.Reason, System.CreatedDate, System.CreatedBy,
System.ChangedDate, System.ChangedBy, System.CommentCount, System.Title, System.BoardColumn,
System.BoardColumnDone, Microsoft.VSTS.Common.StateChangeDate, Microsoft.VSTS.Common.Priority,
Microsoft.VSTS.Common.ValueArea, System.Description, System.Tags FROM WorkItems Where (([System.AreaPath] UNDER
'IT Main\Cloud Infrastructure')) AND (([System.WorkItemType] = 'User Story'))"
VERBOSE: [09:02:40.1795] [Get-TfsWorkItem] Running [Get-TfsTeam] with parameter set 'Simple query' and the
following implicit and explicit arguments:
VERBOSE: [09:02:40.1830] [Get-TfsWorkItem] {"AreaPath" = "IT Main\\Cloud Infrastructure"; "WorkItemType" = ["User
Story"]; "BoardColumnDone" = false; "Ever" = false; "Revision" = 0; "AsOf" = "0001-01-01T00:00:00"; "Fields" =
["System.AreaPath"; "System.TeamProject"; "System.IterationPath"; "System.WorkItemType"; "System.State";
"System.Reason"; "System.CreatedDate"; "System.CreatedBy"; "System.ChangedDate"; "System.ChangedBy";
"System.CommentCount"; "System.Title"; "System.BoardColumn"; "System.BoardColumnDone";
"Microsoft.VSTS.Common.StateChangeDate"; "Microsoft.VSTS.Common.Priority"; "Microsoft.VSTS.Common.ValueArea";
"System.Description"; "System.Tags"]; "TimePrecision" = false; "ShowWindow" = false; "Deleted" = false;
"IncludeLinks" = false; "Team" = ""; "IncludeSettings" = false; "IncludeMembers" = false; "Default" = false}
VERBOSE: [09:02:40.1875] [Get-TfsWorkItem] Get currently connected Collection
VERBOSE: [09:02:40.1875] [Get-TfsWorkItem] GetClient: Getting an instance of
[Microsoft.TeamFoundation.Core.WebApi.TeamHttpClient]
VERBOSE: [09:02:40.1875] [Get-TfsWorkItem] Get currently connected team
VERBOSE: [09:02:40.1971] [Get-TfsWorkItem] Running [Get-TfsTeamProject] with parameter set 'Simple query' and the
following implicit and explicit arguments:
VERBOSE: [09:02:40.2012] [Get-TfsWorkItem] {"AreaPath" = "IT Main\\Cloud Infrastructure"; "WorkItemType" = ["User
Story"]; "BoardColumnDone" = false; "Ever" = false; "Revision" = 0; "AsOf" = "0001-01-01T00:00:00"; "Fields" =
["System.AreaPath"; "System.TeamProject"; "System.IterationPath"; "System.WorkItemType"; "System.State";
"System.Reason"; "System.CreatedDate"; "System.CreatedBy"; "System.ChangedDate"; "System.ChangedBy";
"System.CommentCount"; "System.Title"; "System.BoardColumn"; "System.BoardColumnDone";
"Microsoft.VSTS.Common.StateChangeDate"; "Microsoft.VSTS.Common.Priority"; "Microsoft.VSTS.Common.ValueArea";
"System.Description"; "System.Tags"]; "TimePrecision" = false; "ShowWindow" = false; "Deleted" = false;
"IncludeLinks" = false}
VERBOSE: [09:02:40.2042] [Get-TfsWorkItem] Get currently connected Collection
VERBOSE: [09:02:40.2067] [Get-TfsWorkItem] GetClient: Getting an instance of
[Microsoft.TeamFoundation.Core.WebApi.ProjectHttpClient]
VERBOSE: [09:02:40.2082] [Get-TfsWorkItem] Get currently connected team project

Confirm
Error getting work items (Not Found. )
[Y] Yes  [A] Yes to All  [H] Halt Command  [S] Suspend  [?] Help (default is "Y"): Y
Get-TfsWorkItem : Error getting work items (Not Found. )
At line:1 char:1
+ Get-TfsWorkItem -AreaPath $TeamDetails.AreaPath -WorkItemType "User S ...
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (:) [Get-TfsWorkItem], Exception
    + FullyQualifiedErrorId : VssServiceResponseException,TfsCmdlets.Cmdlets.WorkItem.GetWorkItem

VERBOSE: [09:02:44.7028] [Get-TfsWorkItem] EndProcessing



Looking at the code you have in Github, it may be related to lines 207 to 215 in CSharp/TfsCmdlets/Controllers/WorkItem/GetWorkItemController.cs .... but not 100% sure
